src/tree/azure/GroupingItem.ts

```diff
@@ -101,7 +102,7 @@ export class GroupingItem implements ResourceGroupsItem {
         treeItem.iconPath = this.iconPath;
         treeItem.id = this.id;
 
-        return treeItem;
+        return treeItemState.applyToTreeItem({ ...treeItem, id: this.id });
```

Comment:
An idea of mine is to find the best way to apply this to all tree items, rather than manually applying it in each `getTreeItem` method.

Possible an abstract base class for tree items, but I'm still working on it.